### DISCUSSION TITLE
Freesurfer segmentation

### DISCUSSION THREAD

**Discussion by parinazbabaee** [nipreps/nibabies#299] [Q&A] [answered]
Dear NiBabies Experts,

I am using NiBabies to preprocess resting state fMRI of infants (2-month-old). I have both T1w and T2w images with the following dimensions, respectively: 3, 140, 256, 256 and 3, 156, 256, 256

I used the following code:

nibabies-wrapper docker /Users/parinaz/Desktop/dat /Users/parinaz/Desktop/dat/nibabies participant --participant-label 23083 --age-months 2 -w /Users/parinaz/Desktop/work/ --fs-license-file /Users/parinaz/Documents/license.txt --skip_bids_validation --use-aroma --md-only-boilerplate --stop-on-first-crash --output-spaces MNIInfant:cohort-1 --fs-no-reconall

The analysis finished successfully. However, the output segmentation and registration in the standard space don't look fine. I attached the results. 
segmentation in native space also doesn't look so good, but it is better than the one in standard space.

I want to know if there is any way to improve the result of segmentation and registration in standard space.

Thank you for your consideration,
Parinaz.

<img width="1384" alt="MNIInfant_space" src="https://github.com/nipreps/nibabies/assets/107347044/f31fd98c-53a9-4edd-94f5-fa4ef314bb1e">
<img width="1386" alt="Native_space" src="https://github.com/nipreps/nibabies/assets/107347044/a731c466-a0f2-420a-aa1d-7d118a3e413a">

**Comment by mgxd** [comment, marked as answer]
Hi @parinazbabaee - Automated segmentation is one of the trickier steps to get right in nibabies. I have a few suggestions, ranging from easier to harder that you can try:

- It looks like you are not using FreeSurfer (since you included the `--fs-no-reconall` flag in your command). The fallback is FSL FAST, which from some limited testing is not expected to perform very well in younger subjects due to a lack of tissue contrast. If you remove that and try again, the pipeline will use `infant_recon_all` for this.
- If you have a few gold-standard labeled participants (preferably around the same age), you could leverage them using the `--segmentation-atlases-dir` flag to enable running ANTs JointLabelFusion - more detail on this can be found here: https://nibabies.readthedocs.io/en/stable/faqs.html#multi-atlas-segmentation-with-joint-label-fusion
- You could try using a deep-learning tool, such as https://github.com/DCAN-Labs/BIBSnet, to generate these segmentations, and then feed in these precomputed derivatives via https://nibabies.readthedocs.io/en/stable/faqs.html#leveraging-precomputed-results

**Reply by parinazbabaee** [reply]
Hi Mathias,

Thank you for your kind reply and helpful suggestions. I have already tried using recon-all from the freesurfer; unfortunately, the results didn't improve much. I will try the other two options, which appear to be efficient in fixing my problem. 

Best,
Parinaz.
